[dynamo] Add support for nn.Parameter constructor (part 2)

jansel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #121610
* __->__ #120965
* #121735

This handles the case where the tensor isn't an input.

The changes to dynamo tests are cases where we would previously fall back to eager.



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

jansel committed:
[dynamo] Add support for nn.Parameter constructor (part 2)

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Add support for nn.Parameter constructor (part 2)"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Add support for nn.Parameter constructor (part 2)"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Add support for nn.Parameter constructor (part 2)"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Add support for nn.Parameter constructor (part 2)"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Add support for nn.Parameter constructor (part 2)"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Add support for nn.Parameter constructor (part 2)"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Add support for nn.Parameter constructor (part 2)"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Add support for nn.Parameter constructor (part 2)"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Add support for nn.Parameter constructor (part 2)"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Add support for nn.Parameter constructor (part 2)"


This handles the case where the tensor isn't an input.



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Add support for nn.Parameter constructor (part 2)"


This handles the case where the tensor isn't an input.



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Add support for nn.Parameter constructor (part 2)"


This handles the case where the tensor isn't an input.



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Add support for nn.Parameter constructor (part 2)"


This handles the case where the tensor isn't an input.



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Add support for nn.Parameter constructor (part 2)"


This handles the case where the tensor isn't an input.



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Add support for nn.Parameter constructor (part 2)"


This handles the case where the tensor isn't an input.



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Add support for nn.Parameter constructor (part 2)"


This handles the case where the tensor isn't an input.



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Add support for nn.Parameter constructor (part 2)"


This handles the case where the tensor isn't an input.



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Add support for nn.Parameter constructor (part 2)"


This handles the case where the tensor isn't an input.



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Add support for nn.Parameter constructor (part 2)"


This handles the case where the tensor isn't an input.



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Add support for nn.Parameter constructor (part 2)"


This handles the case where the tensor isn't an input.



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Add support for nn.Parameter constructor (part 2)"


This handles the case where the tensor isn't an input.

The changes to dynamo tests are cases where we would previously fall back to eager.



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

jansel committed:
Update on "[dynamo] Add support for nn.Parameter constructor (part 2)"


This handles the case where the tensor isn't an input.

The changes to dynamo tests are cases where we would previously fall back to eager.



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

yanboliang reviewed (approved):
> The changes to dynamo tests are cases where we would previously fall back to eager.

Just for my learning, why these tests trigger error rather than fall back to eager automatically after this change?

jansel commented:
The previous `unimplemented()`+fallback errors were hiding louder errors later in the function.

jansel committed:
Update on "[dynamo] Add support for nn.Parameter constructor (part 2)"


This handles the case where the tensor isn't an input.

The changes to dynamo tests are cases where we would previously fall back to eager.



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]
